src/components/teas/Teas.tsx

```diff
@@ -1,37 +1,103 @@
 import './Teas.css';
 import Card from '../card/Card';
-import { useParams } from 'react-router-dom';
+import { useParams, useNavigate } from 'react-router-dom';
 import { useState, useEffect } from 'react';
 import { fetchTea } from '../../apiCalls';
+import { Tea } from '../../utils/interface';
+import React from 'react';
 
-function Teas() {
-    // const category: string = useParams().category
-    // const [teaData , setTeaData] = useState()
-
-    // const teaCategory = teaData.filter((tea: Tea) => {
-    //     //Tea category
-    //     return (
-    //         <Card img = {tea.img} name = {tea.name}/>
-    //     )
-    // })
-    // interface Tea {name : string, img :string}
-    // interface Categories { categories: object[]}
-    // let categoryTeas: Categories[] = teaData.filter(() => {
-
-    // })
-    
-    
-    // function displayTeas() {
-            
-        
-    // }
+interface Props {
+    setIsRedirected: React.Dispatch<React.SetStateAction<boolean | undefined>>
+}
+
+function Teas({ setIsRedirected }: Props) {
+    const category = useParams<string>().category
+    const navigate = useNavigate()
+
+    const [teas , setTeas] = useState<Tea[] | null>(null)
+    const initialFavs: Tea[] = JSON.parse(sessionStorage.getItem("favs") || '[]');
+    const [favs, setFavs] = useState<Tea[]>(initialFavs);
+
+    useEffect(() => { 
+        fetchData();
+    }, [category])
+
+    async function fetchData() {
+        if(category === 'favorites') {
+            return setTeas(favs)
+        } else {
+            const fetchedTeaData = await fetchTea();
+            if(!fetchedTeaData) {
+                navigate('*', {replace: true})
+            } else {
+                setIsRedirected(false)
+                setFilteredTeas(fetchedTeaData)
+            }
+        }
+    }
+
+    function setFilteredTeas(fetchedTeaData: any) {
+        const filteredTeaData = fetchedTeaData?.filter((tea: Tea) => tea.type === category)
+        !filteredTeaData.length ? navigate('*', {replace: true}) : setTeas(filteredTeaData);
+    }
+
+    function addFavs(newFav: Tea ) {
+        console.log(newFav)
+        if(favs.includes(newFav)){
+           setFavs(favs.filter(fav => {
+                return fav.slug !== newFav.slug
+            }))
+        }else{
+         setFavs([...favs, newFav])
+        }
+    }
+
+    useEffect(() => {
+        sessionStorage.clear()
+        sessionStorage.setItem("favs", JSON.stringify(favs));
+        fetchData()
+    }, [favs])
+
+    const teaCards = teas?.map((tea: Tea) => {
+        return (
+            <Card
+                tea={tea}
```

Comment:
This might be something with TS that I'm missing, but it feels strange to me that you're passing the entire `tea` object AND each individual key as separate props.